### PR TITLE
Add driver for the Agilent N9000A

### PR DESCRIPTION
**Pull request by edumur**

### CODE REVIEW

**Comment by jenshnielsen** [comment]
@edumur Thnaks for your pull request. To get the job to pass please mark the notebook as non executable as documented here 
https://qcodes.github.io/Qcodes/examples/writing_drivers/Creating-Instrument-Drivers.html#Documentation 

Please also ensure that there is a empyty `__init__.py` file in the Agilent directory

**Comment by edumur** [comment]
Hi,

I was sure to miss something.
I implemented your requests.

**Comment by jenshnielsen** [comment]
It looks like some unrelated HF2LI changes made it in. Could you revert those

**Comment by edumur** [comment]
My bad.
Commit reverted.

File: docs/examples/Agilent_N9000A.ipynb [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import qcodes as qc\n",
+    "import sys\n",
+    "sys.path.append('../../')\n",
```
Comment (jenshnielsen) [inline docs/examples/Agilent_N9000A.ipynb:51]: Please don't use path modifying tricks in the documentation. qcodes_contrib_drivers is a package that can be installed normally. E.g. `pip install .` or `pip install -e .` from the root folder of the repository (or when this makes it into a release `pip install qcodes_contrib_drivers`

File: qcodes_contrib_drivers/drivers/Agilent/Agilent_N9000A.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+
+    def power_parser(self, val: str, output: str) -> float:
+        """
+        Parse the reply from a strin containing 'power, psd' to float depending
+        of the output.
+
+        Args:
+            val (str): Reply of the power spectral analyzer.
+            output (str): Desired output format.
```
Comment (jenshnielsen) [inline qcodes_contrib_drivers/drivers/Agilent/Agilent_N9000A.py:94]: We normally don't bother with including types in the docstrings when they are already in the signature. The types in the signature are checked by mypy and are therefor far more likely to be correct

File: qcodes_contrib_drivers/drivers/Agilent/Agilent_N9000A.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+class Agilent_N9000A(VisaInstrument):
+    """
+    This is the QCoDeS python driver for the Agilent CXA N9000A power spectrum
+    analyzer.
+    """
+    
+    
+    
+    def __init__(self, name       : str,
```
Comment (jenshnielsen) [inline qcodes_contrib_drivers/drivers/Agilent/Agilent_N9000A.py:17]: You have a lot of blank lines here. Most python style guids would recommend only one or two blank lines depending on the situation

File: docs/examples/Agilent_N9000A.ipynb (hunk 1 above)
Comment (edumur) [reply docs/examples/Agilent_N9000A.ipynb]: I modified the import.

File: qcodes_contrib_drivers/drivers/Agilent/Agilent_N9000A.py (hunk 2 above)
Comment (edumur) [reply qcodes_contrib_drivers/drivers/Agilent/Agilent_N9000A.py:90]: If you leave it to me, I will keep it like this.
However I can remove them if ask.

File: qcodes_contrib_drivers/drivers/Agilent/Agilent_N9000A.py (hunk 3 above)
Comment (edumur) [reply qcodes_contrib_drivers/drivers/Agilent/Agilent_N9000A.py:16]: Normally corrected, please tell me if that format suits you better.

File: qcodes_contrib_drivers/drivers/Agilent/Agilent_N9000A.py (hunk 2 above)
Comment (jenshnielsen) [reply qcodes_contrib_drivers/drivers/Agilent/Agilent_N9000A.py:90]: It's fine either way

**Review by jenshnielsen — APPROVED** [review]

**Pull request merged**